FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,276 @@
+                    R.dimen.fluentui_contextual_command_bar_default_item_padding_vertical
+            )
+            itemPaddingHorizontal = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_default_item_padding_horizontal
+            )
+        }
+
+        setDismissButtonVisible(showDismiss)
+    }
+
+    private fun addDismissButton() {
+        // Container
+        dismissButtonContainer = LinearLayout(context).apply {
+            orientation = LinearLayout.HORIZONTAL
+        }
+        addView(dismissButtonContainer)
+        (dismissButtonContainer!!.layoutParams as LayoutParams).apply {
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = ViewGroup.LayoutParams.WRAP_CONTENT
+            gravity = Gravity.END
+        }
+
+        // Dismiss gradient gap
+        val gradientGap = View(context).apply {
+            background = ContextCompat.getDrawable(
+                    context,
+                    R.drawable.contextual_command_bar_dismiss_button_gap_background
+            )
+        }
+        dismissButtonContainer!!.addView(gradientGap)
+        gradientGap.layoutParams.apply {
+            width = resources.getDimensionPixelSize(R.dimen.fluentui_contextual_command_bar_dismiss_gap_width)
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+        }
+
+        // Dismiss button
+        itemPaddingVertical = resources.getDimensionPixelSize(
+                R.dimen.fluentui_contextual_command_bar_default_item_padding_vertical
+        )
+        itemPaddingHorizontal = resources.getDimensionPixelSize(
+                R.dimen.fluentui_contextual_command_bar_default_item_padding_horizontal
+        )
+        dismissButton = ImageView(context).apply {
+            setPaddingRelative(
+                    itemPaddingHorizontal,
+                    itemPaddingVertical,
+                    itemPaddingHorizontal,
+                    itemPaddingVertical
+            )
+            setBackgroundColor(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissBackgroundColor
+            ))
+            imageTintList = ColorStateList.valueOf(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissIconTintColor
+            ))
+
+            setOnClickListener {
+                dismissListener?.invoke()
+            }
+        }
+        dismissButtonContainer!!.addView(dismissButton)
+        dismissButton!!.layoutParams.apply {
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_button_width
+            )
+        }
+
+        bringChildToFront(dismissButtonContainer)
+    }
+
+    private fun setDismissButtonVisible(visible: Boolean) {
+        if (dismissIcon == 0) {
```

Comment:
would it make sense to use the setting of the dismiss icon or not to determine whether or not we show dismiss? if set we show it, if not we don't? then we don't need to setShowDismiss and setDismissIcon